nibabel/gifti/gifti.py

```diff
@@ -386,14 +385,44 @@ def metadata(self):
         return self.meta.metadata
 
 
-class GiftiImage(FileBasedImage, xml.XmlSerializable):
+class GiftiImage(xml.XmlBasedImage):
+    """
+    The Gifti spec suggests using the following suffixes to your
+    filename when saving each specific type of data:
+
+    .gii
+        Generic GIFTI File
+    .coord.gii
+        Coordinates
+    .func.gii
+        Functional
+    .label.gii
+        Labels
+    .rgba.gii
+        RGB or RGBA
+    .shape.gii
+        Shape
+    .surf.gii
+        Surface
+    .tensor.gii
+        Tensors
+    .time.gii
+        Time Series
+    .topo.gii
+        Topology
+
+    The Gifti file is stored in endian convention of the current machine.
+    """
     valid_exts = ('.gii',)
     files_types = (('image', '.gii'),)
 
     def __init__(self, header=None, extra=None, file_map=None, meta=None,
                  labeltable=None, darrays=None, version="1.0"):
-        FileBasedImage.__init__(self, header=header, extra=extra,
-                                file_map=file_map)
+        super(GiftiImage, self).__init__(header=header, extra=extra,
+                                         file_map=file_map)
+        # placed here temporarily for git diff purposes
+        from .parse_gifti_fast import GiftiImageParser
+        GiftiImage.parser = GiftiImageParser
```

Comment:
Yep. the image and parsing code refer to each other. I think it helps to keep them as separate files, to keep files short. So this is necessary.

The same design is used in the Cifti code. There, it's even more helpful to keep the object defs and parsing code separate, as both files are relatively long.